pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -616,9 +616,50 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 		changes.KillPod = true
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.SidecarLifecycle) {
```

Comment:
we had faced a very complicated race condition during pressure test in our private branch of sidecar impl that you might want to handle:
- Pod has 1 main container and 2 sidecars
- main container finishes
- computePodActions determines that we should kill the 2 sidecars
- we issue kill command to the 2 sidecar containers
- 1st sidecar exited and triggers pod sync
- 2nd sidecar exited and event is queued
- we reconcile container statuses and found the 2 sidecar are both exited
- we kill pod by killing sandbox
- we started to process sidecar exit event for 2nd sidecar
- we don't have a pod sandbox now and podSandboxChanged() returned that we should create sandbox
- we somehow failed to create sandbox (possibly) because the pod cgroup is already being teared down
- subsequent computePodActions() calls will fail with same reason 

https://github.com/zhan849/kubernetes/commit/540971d21764692d5a33c029c5b1a938dcb444b5#diff-8da51ab012ecbdf2b013ecbcf1c37033R584 for your reference

Reply:
Yes, we hit the same race in our testing as well.

Reply:
thank, that's really useful, I will have a look and try to implement it!

Reply:
Has the above race condition been addressed ?
Thanks

Reply:
Yeah, I introduced a check here https://github.com/kubernetes/kubernetes/pull/80744/commits/c306bcd8fc94f2073a785f16c47056f1e0717801#diff-8da51ab012ecbdf2b013ecbcf1c37033R533-R538 
This should prevent this behaviour from happening by stopping pod sandbox recreation during the sidecar shutdown sequence.